src/core/TimePos.cpp

```diff
@@ -72,7 +72,12 @@ TimePos TimePos::quantize(float bars) const
 	//Offset from the lower position
 	int offset = m_ticks % interval;
 	//1 if we should snap up, 0 if we shouldn't
-	int snapUp = offset / (interval / 2);
+	// Ternary expression is making sure that the snap happens in the direction to
+	// the right even if m_ticks is negative and the offset is exactly half-way
+	// More details on issue #5840 and PR #5847
+	int snapUp = (offset == -(interval / 2))
```

Comment:
This won't wotk properly if `interval` is odd, please multiply both sides by 2.

Reply:
Oh true, integer division, my bad 😬 
Will fix once I get home

Reply:
It was there from the start as well, so blame me instead! 😅

Reply:
Fixed on last commit! The equation might look less obvious, think I should've added a comment?